titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/jobmanager/replicator/GrpcJobReplicatorEventStream.java

```diff
@@ -149,12 +149,13 @@ public GrpcJobReplicatorEventStream(JobManagementClient client,
                     coreEvent = toJobCoreEvent(job);
                     break;
                 case TASKUPDATE:
+                    //TODO(fabio): handle task moved when it is added to the gRPC API definition
                     com.netflix.titus.grpc.protogen.Task task = event.getTaskUpdate().getTask();
-                    Job<?> taskJob = lastSnapshot.getJobs().stream().filter(j -> j.getId().equals(task.getJobId())).findFirst().orElse(null);
-                    if (taskJob != null) {
-                        Task coreTask = V3GrpcModelConverters.toCoreTask(taskJob, task);
+                    Optional<Job<?>> taskJob = lastSnapshot.findJob(task.getJobId());
```

Comment:
can you rename this to something like `taskJobOpt` and use `taskJob` as a local variable if it is present?